src/bp/core/services/dialog/flow/service.ts

```diff
@@ -132,12 +129,16 @@ export class FlowService {
 
         if (await this.ghost.forBot(botId).fileExists(FLOW_DIR, flowPath)) {
           const flow = await this.parseFlow(botId, flowPath)
-          const flowWithParents = this.addParentsToFlows([flow])[0]
 
-          this._flowCache.upsertFlow(botId, flowWithParents)
+          this._flowCache.upsertFlow(botId, flow)
         } else {
           this._flowCache.deleteFlow(botId, flowPath)
         }
+
+        const flows = this._flowCache.get(botId)
+        const flowsWithParents = this.addParentsToFlows(flows)
+
+        this._flowCache.set(botId, flowsWithParents)
```

Comment:
Let's push this improvement one step further. Parent flows are only used by the NDU, which represent less than 5% of the bots (probably less). We could just skip the addParents altogether for non-ndu flows: 
```js
if(this._isOneFlow(botId){
   this._flowCache.set(botId, flowsWithParents)
} else {
   this._flowCache.set(botId, flows)
}
```

Reply:
or we could drop ndu support ;)